src/tools/rados/rados.cc

```diff
@@ -3312,27 +3376,26 @@ static int rados_tool_common(const std::map < std::string, std::string > &opts,
       }
     }
   } else if (strcmp(nargs[0], "set-alloc-hint") == 0) {
-    if (!pool_name || nargs.size() < 4) {
+    if (!pool_name || nargs.size() < (obj_name ? 3 : 4)) {
       usage(cerr);
       return 1;
     }
     string err;
-    string oid(nargs[1]);
-    uint64_t expected_object_size = strict_strtoll(nargs[2], 10, &err);
+    uint64_t expected_object_size = strict_strtoll(nargs[obj_name ? 1 : 2], 10, &err);
```

Comment:
how about:
```c++
size_t arg_index = 1;
string oid;
if (obj_name) {
  oid = *obj_name;
} else {
  oid = nargs[arg_index++];
}
uint64_t expected_object_size = strict_strtoll(nargs[arg_index++], 10, &err);
// ...
```

Reply:
Stayed with the ternary operator to be consistent with other commands.

Reply:
i just don't like the magic numbers. difficult to reason about and maintain.